p2p/state.py

```diff
@@ -138,15 +120,9 @@ def idle_peers(self) -> List[ETHPeer]:
             self.logger.debug("Ignoring %s msg while doing a StateSync", cmd)
 
     async def _cleanup(self):
-        start_at = time.time()
-        # Wait at most 1 second for pending peers to finish.
-        self.logger.info("Waiting for %d running peers to finish", len(self._running_peers))
-        while time.time() < start_at + 1:
-            if not self._running_peers:
-                break
-            await asyncio.sleep(0.1)
-        else:
-            self.logger.info("Waited too long for peers to finish, exiting anyway")
+        # We don't need to cancel() anything, but we yield control just so that the coroutines we
+        # run in the background notice the cancel token has been triggered and return.
+        await asyncio.sleep(0)
```

Comment:
Isn't the promise of cleanup that all subservices are shut down by the time it returns?

If I understand that right, then I would expect a pattern like `await asyncio.gather(*[subservice.cleaned_up.wait() for subservice in subservices])` here.

(plus the same thing in p2p/chain)

Reply:
StateDownloader has no sub-services. It only runs a few coroutines in the background and I plan to have a new method on `BaseService` so that we can easily keep track of those and wait for them to return on `_cleanup()`, but right now there's no way to do that